source/util/parse_number.h

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LIBSPIRV_UTIL_PARSE_NUMBER_H_
+#define LIBSPIRV_UTIL_PARSE_NUMBER_H_
+
+#include <functional>
+#include <string>
+#include <tuple>
+
+#include "util/hex_float.h"
+
+namespace spvutils {
+
+// TODO(qining): Rename those types and update the blurbs to decouple from
```

Comment:
In the assembler context, values of this type help keep track of the type of a value *referenced* by a given Id. 
But by the time we're in the parsing utilities, we already know the target type (float, int, uint) and the bit width.
The data here corresponds to spv_number_kind_t and bit width.